python/pyarrow/array.pyx

```diff
@@ -68,6 +68,14 @@ cdef class Array:
         values = array_format(self, window=10)
         return '{0}\n{1}'.format(type_format, values)
 
+    def __richcmp__(Array self, Array other, int op):
```

Comment:
To head off semantic issues around `a == b` (as compared with NumPy behavior) we might put this logic in a `def equals(self, other): ...`